File: regexlib/javascript_test_file/regexlib_1125.js
/* 1125
 * (href=|url|import).*[\'"]([^(http:)].*css)[\'"]
 * POLYNOMIAL
 * nums:3
 * POLYNOMIAL AttackString:""+"href="*5000+"! _1SLQ_2"
 */
var REGEX = /(href=|url|import).*[\'"]([^(http:)].*css)[\'"]/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + 'href='.repeat(i*10000) + '! _1SLQ_2'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 10000 + ": " + time_cost+" ms")
}

File: regexlib/2021-5-15/python_re2_test_file/regexlib_1125.py
# 1125
# (href=|url|import).*[\'"]([^(http:)].*css)[\'"]
# POLYNOMIAL
# nums:5
# POLYNOMIAL AttackString:""+"href="*5000+"! _1SLQ_2"

import re2 as re
from time import perf_counter

regex = """(href=|url|import).*[\'"]([^(http:)].*css)[\'"]"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "href=" * i * 10000 + "! _1SLQ_2"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *10000}: took {DURATION} seconds!")